pandas/core/indexes/datetimelike.py

```diff
@@ -806,6 +806,25 @@ def _fast_union(self, other, sort=None):
         else:
             return left
 
+    def _union(self, other, sort):
```

Comment:
other can a Union[DTI, TDI]

Reply:
@simonjayhawkins what's the preferred way of typing as `type(self)`?

Reply:
def _union(self: _T, other: Type[_T], sort):

Reply:
where does _T come from?  I dont see this pattern anywhere else in the code

Reply:
i vote for merging and adding annotations in follow-up

Reply:
followup ok